codice/ddf
# DDF-4107, DDF-4106 Adds prettier formatter to Intrigue

#### What does this PR do?
- Adds prettier formatter to Intrigue
- Migrates source components

#### Who is reviewing it? 
@Schachte 
@brendan-hofmann
@rzwiefel
@tbatie
@vinamartin
@djblue 

#### Note to Reviewers
Check out the prettier config and tell me if you disagree 😄 
```
{
    singleQuote: true,
    trailingComma: 'es5'
}
```
I put some files in the ignore because prettier said formatting them would result in possible behavior changes.

#### How should this be tested?
To test the source migration make a few sources and verify it works as before.  When sources go down, things should update in the UI.  Make a source action and verify it shows up in the UI and works as intended.

To test the prettier formatter, pull down the changes.  Run `yarn build` and verify it passes.  Make a change to a file that goes against the formatter.  Run `yarn build` and verify it fails.

Right now I only have it setup to verify changes versus master for building, but we could swap it to look at all files.

#### Any background context you want to provide?
#### What are the relevant tickets?
https://codice.atlassian.net/projects/DDF/issues/DDF-4106
https://codice.atlassian.net/projects/DDF/issues/DDF-4107

<img width="1409" alt="screen shot 2018-08-29 at 9 53 46 pm" src="https://user-images.githubusercontent.com/11984853/44830335-13e62100-abd6-11e8-9d72-3ebe18345ace.png">
